### ISSUE TITLE
Searching tests infinitively slow but not always for Spring projects

### ISSUE THREAD

**Issue by ivorobioff** [microsoft/vscode-java-test#1203]
When I try to run all tests for one of the projects in my workspace from the "Testing" tab it starts searching tests too long (I waited about 15 min max. and gave it up). However, after running all tests on my another project from the same workspace it took several seconds to find tests and all tests were executed ok. The interesting thing is that after this, when I run again all tests for the first project I mentioned above all tests were found this time quickly and executed.

Here's my info

```
Version: 1.56.2 (user setup)
Commit: 054a9295330880ed74ceaedda236253b4f39a335
Date: 2021-05-12T17:13:13.157Z
Electron: 12.0.4
Chrome: 89.0.4389.114
Node.js: 14.16.0
V8: 8.9.255.24-electron.0
OS: Windows_NT x64 10.0.19042
```

Please let me know if anything else is needed.

**Comment by jdneo** [comment]
Is there any way to stable reproduce it?

**Comment by baincd** [comment]
I have also experienced this behavior.  This is on a Multi-module Spring Boot project using Maven.  I have not yet been able to reliably reproduce it.

What happens is:
- Test sidebar > Run all tests (never finished searching for tests)
- Cancel
- Run all the tests individually from the Explorer sidebar (by module or package) - they all run fine
- Test sidebar > Run all tests (finds and runs tests quickly)

**Comment by jdneo** [comment]
Reopen this since @baincd provides steps to repro. Will take a look when I have time.

Thanks.

**Comment by ghartz-hz** [comment]
Just a datapoint, if it helps -- this has been a problem for me since early 2021. I'm not positive when it showed up, but it was at least 4-5 months ago. 

It only seems to happen at the project root -- going down to a package level (even a package with a lot of tests) finds and starts them instantly. But even a tiny project with a single test, when run at the root of the tree, never actually returns.

**Comment by jdneo** [comment]
I tried the https://github.com/spring-projects/spring-petclinic project but not able to reproduce this.

Anyway, we are close to release branding-new extension really soon. ([adopting the VS Code new testing API](https://github.com/microsoft/vscode-java-test/issues/1162)), hope that at that time this problem could be resolved.

**Comment by jdneo** [comment]
We released 0.31.0 today which consumed the VS Code's new testing APIs. This release also includes bunch of refactorings. I'm closing this issue for now since so far I cannot find a way to repro this issue.

But feel free to leave comments here if it still occurs in the new extension. Thanks!
